src/openrct2-ui/windows/TrackDesignPlace.cpp

```diff
@@ -543,20 +543,20 @@ static void window_track_place_draw_mini_preview_track(
         const rct_preview_track* trackBlock = trackBlockArray[trackType];
         while (trackBlock->index != 255)
         {
-            int16_t x = origin.x;
-            int16_t y = origin.y;
-            map_offset_with_rotation(&x, &y, trackBlock->x, trackBlock->y, rotation);
+            auto rotatedAndOffsetTrackBlock = map_offset_with_rotation(
```

Comment:
This isn't correct a `rct_preview_track*` coordinates are in `CoordsXY`. I think this would be simplified if we did the following:

```
CoordsXY trackCoords = origin + CoordsXY{trackBlock->x, trackBlock->y}.Rotate(rotation);
```